packages/sandcastle/src/App.tsx

```diff
@@ -241,11 +247,46 @@ function App() {
           committedCode: action.code ?? state.code,
           committedHtml: action.html ?? state.html,
           runNumber: state.runNumber + 1,
+          dirty: false,
+        };
+      }
+      case "resetDirty": {
+        return {
+          ...state,
+          dirty: false,
         };
       }
     }
   }, initialState);
 
+  useEffect(() => {
+    const host = window.location.host;
+    let envString = "";
+    if (host.includes("localhost") && host !== "localhost:8080") {
+      // this helps differentiate tabs for local sandcastle development or other testing
+      envString = `${host.replace("localhost:", "")} `;
+    }
+
+    const baseTitle = "Cesium Sandcastle";
```

Comment:
```suggestion
    const baseTitle = "CesiumJS Sandcastle";
```

or possibly even just the following due to all the other info?

```suggestion
    const baseTitle = "Sandcastle";
```

Reply:
I'm looking at what we do on our other products and marketing website. Unless you see an issue, let's follow suit with what https://ion.cesium.com/ does—`<Page Title> | Cesium ion`, such as `Stories | Cesium ion` or `My Assets | Cesium ion`.

So for Sandcastle, I'm thinking `Sandcastle | CesiumJS` by default, and `<Gallery Item Title> | Sandcastle | CesiumJS`, such as `3D Models | Sandcastle | CesiumJS` or `Geometry and Appearances | Sandcastle | CesiumJS`, where appropriate.